### PR TITLE
ELPDES-92: Featured text component

### PR DESCRIPTION
**Pull request by pauzim**
Designs: https://www.figma.com/file/xy6otn2JWHNdF70Tuq0UbS/TACO-Design-System-%5BDRAFT%5D?node-id=3896-36061&t=kfHLqXWPj8lXwNcw-0

I've separated `icon` and `imgSrc` props because of AC that we have to clip and fit icons if they are images. Not sure if this logic doesn't complicate this component too much. If for some reason both `icon` and `imgSrc` are provided, `imgSrc` has priority.

### CODE REVIEW

**Review by insanepl — COMMENTED** [review]
Remove package.lock and use pnpm to install. Might be some issues so wait for this after we merge (https://github.com/elpassion/flounder-2/pull/6)

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  variant,
+  size,
+  align,
+  linkedText,
+  linkedUrl,
+  ...props
+}) => {
+  const featuredTextVariants = {
+    text: "",
+    iconTop: "flex-col gap-y-4",
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:21]: We can think about gaps already in figma tokens maybe? so we can customize spacing?

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    center: "items-center text-center",
+  };
+
+  const featuredTextTitleSizes = {
+    sm: "text-base",
+    md: "text-lg",
+    lg: "text-2xl",
+  };
+
+  const learnMoreSizeVariants = {
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:36]: Same with the gaps and tokens I guess

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  return (
+    <>
+      {isIconVariant && (
+        <div
+          className={classNames(
+            "flex flex-shrink-0 items-center justify-center rounded-md bg-neutral-50",
+            iconWrapperSizes[size]
+          )}
+        >
+          <span
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:138]: How abouit if we just put {icon} here and pass svg, img, or font-icon from the to (storybook) as I did with NavTab

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            className={classNames("font-icons", iconSizes[size])}
+            dangerouslySetInnerHTML={{ __html: `${icon};` }}
+          />
+        </div>
+      )}
+
+      {isImageVariant && (
+        <img
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:146]: Can also be ditched if we just pass it as reactnode

File: packages/design-system/src/stories/molecules/FeaturedText.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  args: {
+    title: "Featured component title",
+    content:
+      "Your community is there for your and they want to help you grow but even though they want to.",
+    linkedText: "Learn more",
+    size: "sm",
+    align: "left",
+    variant: "iconTop",
+    linkedUrl: "https://google.com",
+    imageSrc: "images/rectangle.png",
```
Comment (insanepl) [inline packages/design-system/src/stories/molecules/FeaturedText.stories.tsx:82]: Let 's put image/move images like this or avatar to storybook public, so it's not part of the lib: https://storybook.js.org/docs/react/configure/images-and-assets

**Review by insanepl — CHANGES_REQUESTED** [review]

File: packages/design-system/src/components/FeaturedText/index.tsx (hunk 4 above)
Comment (pauzim) [reply packages/design-system/src/components/FeaturedText/index.tsx:146]: It was my first approach but I stucked when it cames to fit passed img to specific size eg clip rectangle size to square etc 🤔

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 6]
```diff
@@ -40,9 +41,9 @@ export const FeaturedText: React.FC<FeaturedTextProps> = ({
   };
 
   const learnMoreIconSizes = {
-    sm: "text-base",
-    md: "text-base",
-    lg: "text-2xl",
+    sm: "w-4 h-4",
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:43]: w-4 aspect-square and w-6 aspect-square should work

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 7]
```diff
@@ -94,9 +95,7 @@ export const FeaturedText: React.FC<FeaturedTextProps> = ({
           )}
         >
           <a href={linkedUrl}>{linkedText}</a>
-          <span className={classNames("font-icons", learnMoreIconSizes[size])}>
-            &#xeb12;
-          </span>
+          <ArrowRight className={classNames(learnMoreIconSizes[size])} />
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:103]: no need classNames here, Remember hat we're going to export all those components as a lib, so it's best if we stick to import react if possible

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
+import classNames from "classnames";
+import {
+  FeaturedTextProps,
+  TFeaturedTextSize,
+  TFeaturedTextVariant,
+} from "./Featuredtext.interface";
+import { ReactComponent as ArrowRight } from "../../svgs/arrow-right.svg";
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:7]: Take a look how I handled Check User etc SVG. Let's use it this way, I found an issue while published to npm with those svgs :)

**Review by insanepl — CHANGES_REQUESTED** [review]
Add Arrow icon as RectCompoennt in svg folder, take a look at the main branch

File: packages/design-system/src/components/FeaturedText/index.tsx (hunk 8 above)
Comment (pauzim) [reply packages/design-system/src/components/FeaturedText/index.tsx]: done!

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  const isIconVariant = variant !== "text" && icon;
+
+  const getFeaturedTextStyles = ({
+    size,
+    variant,
+  }: {
+    size: TFeaturedTextSize;
+    variant: TFeaturedTextVariant;
+  }): string => {
+    switch (size) {
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:64]: Maybe we should use md as default size?

File: packages/design-system/src/svgs/ArrowRight.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
+const ArrowRight = ({ className }: { className?: string }) => {
+  return (
+    <svg
+      width="24"
+      height="24"
```
Comment (insanepl) [inline packages/design-system/src/svgs/ArrowRight.tsx:5]: Remove

**Review by insanepl — CHANGES_REQUESTED** [review]

File: packages/design-system/src/stories/molecules/FeaturedText.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,86 @@
+import { ComponentStory } from "@storybook/react";
+import React from "react";
+import { FeaturedText as FeaturedTextComponent } from "../../components/FeaturedText";
+import { FeaturedTextProps } from "../../components/FeaturedText/Featuredtext.interface";
+
+export const FeaturedText: ComponentStory<React.FC<FeaturedTextProps>> = ({
+  ...props
+}) => <FeaturedTextComponent {...props} />;
+
+const StoryIcon: React.FC<{ src: string }> = ({ src }) => (
```
Comment (insanepl) [inline packages/design-system/src/stories/molecules/FeaturedText.stories.tsx:10]: Move it to storybook utils.tsx, i made new file with control, icons and heleprs like this to keep stories clean :)

Comment (insanepl) [reply packages/design-system/src/stories/molecules/FeaturedText.stories.tsx:10]: OHh I see there's more icons under it, maybe you can use the icons we alreayd got in helpers then :)

File: packages/design-system/src/svgs/ArrowRight.tsx [hunk 12]
```diff
@@ -0,0 +1,20 @@
+const ArrowRight = ({ className }: { className?: string }) => {
```
Comment (insanepl) [inline packages/design-system/src/svgs/ArrowRight.tsx:1]: Pull main and use RightArrowSvg componen I already added, so we avoid duplocate code :) Also change import in compoennt you used it, as I checked it's literally the same code just formatted for JSX

**Review by insanepl — CHANGES_REQUESTED** [review]

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 13]
```diff
@@ -0,0 +1,113 @@
+import classNames from "classnames";
+import RightArrowSvg from "../../svgs/RightArrowSvg";
+
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:3]:
```suggestion
```

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+import {
+  FeaturedTextProps,
+  TFeaturedTextSize,
+  TFeaturedTextVariant,
+} from "./Featuredtext.interface";
+
+export const FeaturedText: React.FC<FeaturedTextProps> = ({
```
Comment (insanepl) [inline packages/design-system/src/components/FeaturedText/index.tsx:10]:
```suggestion
export const FeaturedText: FC<FeaturedTextProps> = ({
```

**Review by insanepl — CHANGES_REQUESTED** [review]

File: packages/design-system/src/components/FeaturedText/index.tsx [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+        textAlignment
+      )}
+    >
+      {isIconVariant && (
+        <div className={classNames("flex-shrink-0", iconWrapperSizes[size])}>
+          {icon}
+        </div>
+      )}
+      <div className={classNames("flex flex-col", textAlignment)}>
+        <h6
```
[the remark this answers is not in the document]
Comment (pauzim) [reply packages/design-system/src/components/FeaturedText/index.tsx:90]: Task added: https://elpassion.atlassian.net/jira/software/projects/ELPDES/boards/246?selectedIssue=ELPDES-158

**Review by insanepl — APPROVED** [review]

**Pull request merged**